### ISSUE TITLE
Add upsert support for Postgresql 

### ISSUE THREAD

**Issue by kyri-petrou** [awslabs/aws-data-wrangler#807]
**Is your idea related to a problem? Please describe.**
Being able to upsert to a postgresql table is very often desired for replacing individual records in a table.

**Describe the solution you'd like**
The same or similar functionality as [the one recently added for Mysql](https://github.com/awslabs/aws-data-wrangler/pull/613) would be perfect!

I'm happy to contribute if needed but I might need some help with the testing.

**Comment by jaidisido** [comment]
This has now been addressed in merged PR #811 and can be tested:

```
pip uninstall awswrangler -y
pip install git+https://github.com/awslabs/aws-data-wrangler.git@main
```

until it's available in the next release. Many thanks to @maxispeicher for the work

**Comment by krspetrou** [comment]
Awesome, thank you very much!
